### DISCUSSION TITLE
Issues with FieldArray and Material-UI Select

### DISCUSSION THREAD

**Discussion by deviprsd** [react-hook-form/react-hook-form#5686] [Q&A]
I encountered an issue where when `react-hook-form` tries to focus on the first form element, and if it is a `SelectInput` from `@material-ui/core/Select/SelectInput.js`, focusing on it causes an error `Cannot read property 'focus' of null`.

There is an ongoing fix for it https://github.com/mui-org/material-ui/pull/23302 but it is set to release with **v5** which is currently in **alpha** stage. I would like to not use `pre-release` versions for a production app. Here is a reproduction https://codesandbox.io/s/inspiring-benz-elkzt

Is there a way to handle this error or remove auto focus for using `react-hook-form` library?

**Comment by rvanpraet** [comment]
You're already using a Controller so no need to register the field using inputRef. If you remove this piece your code will work:

<img width="570" alt="Screen Shot 2021-06-18 at 18 48 50" src="https://user-images.githubusercontent.com/43333514/122593423-f3790100-d065-11eb-9d39-f168fc27b191.png">

**Reply by deviprsd** [reply]
I'm aware that removing the inputRef fixes the issue, but then it won't be able to focusOnError. I mean in all technicality it is `material-ui` problem for not having a null check, but I guess I was looking for more control of autoFocus passed to the developer
